sdk/src/types/block/context_input/mod.rs

```diff
@@ -32,40 +44,80 @@ impl ContextInput {
     /// Returns the context input kind of a `ContextInput`.
     pub fn kind(&self) -> u8 {
         match self {
+            Self::Commitment(_) => CommitmentContextInput::KIND,
+            Self::BlockIssuanceCredit(_) => BlockIssuanceCreditContextInput::KIND,
             Self::Reward(_) => RewardContextInput::KIND,
         }
     }
 
+    /// Checks whether the context input is a [`CommitmentContextInput`].
+    pub fn is_commitment(&self) -> bool {
```

Comment:
Ok I'm super annoying now but in other modules we alternate the is_ and the as_ methods 😬

Reply:
Not sure what do you mean, for me the `is_` should return a `bool` and the `as_` should  cast, which is what we do here 🤔

Reply:
I mean in the order. We do is_ then as_ for each variant

Reply:
😆 Ok I'll change it